server/channels/app/content_flagging.go

```diff
@@ -429,3 +437,277 @@ func (a *App) GetPostContentFlaggingPropertyValues(postId string) ([]*model.Prop
 
 	return propertyValues, nil
 }
+
+func (a *App) PermanentDeleteFlaggedPost(rctx request.CTX, actionRequest *model.FlagContentActionRequest, reviewerId string, flaggedPost *model.Post) *model.AppError {
+	// when a flagged post is removed, the following things need to be done
+	// 1. Hard delete corresponding file infos
+	// 2. Hard delete file infos associated to post's edit history
+	// 3. Hard delete post's edit history
+	// 4. Hard delete the files from file storage
+	// 5. Hard delete post's priority data
+	// 6. Hard delete post's post acknowledgements
+	// 7. Hard delete post reminders
+	// 8. Scrub the post's content - message, props
+
+	commentBytes, jsonErr := json.Marshal(actionRequest.Comment)
+	if jsonErr != nil {
+		return model.NewAppError("PermanentlyRemoveFlaggedPost", "app.content_flagging.permanently_delete.marshal_comment.app_error", nil, "", http.StatusInternalServerError).Wrap(jsonErr)
+	}
+	// Storing marshalled content into RawMessage to ensure proper escaping of special characters and prevent
+	// generating unsafe JSON values
+	commentJsonValue := json.RawMessage(commentBytes)
+
+	status, appErr := a.GetPostContentFlaggingStatusValue(flaggedPost.Id)
+	if appErr != nil {
+		return appErr
+	}
+
+	if status == nil {
+		return model.NewAppError("removeFlaggedPost", "api.content_flagging.error.post_not_flagged", nil, "", http.StatusBadRequest)
+	}
+
+	statusValue := strings.Trim(string(status.Value), `"`)
+	if statusValue != model.ContentFlaggingStatusPending && statusValue != model.ContentFlaggingStatusAssigned {
+		return model.NewAppError("removeFlaggedPost", "api.content_flagging.error.post_not_in_progress", nil, "", http.StatusBadRequest)
+	}
+
+	editHistories, appErr := a.GetEditHistoryForPost(flaggedPost.Id)
```

Comment:
Not related to this PR but now I wonder what if someone caused a data spillage, but that person edited the message after it's being flagged, is there a way for content reviewers to see previous message history?

Reply:
Agreed on not blocking, raised here: [hub](https://hub.mattermost.com/pde/pl/8te8dywnnpycdp49nz4sye9kfc)